plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal.hs

```diff
@@ -252,12 +252,13 @@ cabalRules recorder plId = do
                 -- We don't support the cabal version, this should not be an error, as the
                 -- user did not do anything wrong. Instead we cast it to a warning
                 regex = "Unsupported cabal-version [0-9]+.[0-9]*"
+                unsupportedCabalHelpText = "\nThe used cabal version is not fully supported by hls. This means that some functionallity might not work as expected.\nIf you face any issues try to downgrade to a supported cabal version."
```

Comment:
We can get the set of supported versions from https://github.com/haskell/haskell-language-server/blob/master/plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/Completion/Data.hs#L33 and display it here for the best UX.

You can pull out the lower bound into its own dedicated variable, e.g. 

```haskell
lowestSupportedCabalVersion = CabalSpecV2_2
```

and then use that. Or pull out the list of supported cabal versions... Either way is fine.